### PR TITLE
Improved user registration

### LINKED ISSUE #922: Improve user registration
Currently if during registration on the website, the wrong email address is entered. It's not possible to register again as there already is an entry in the NewUser table.
This results in a PDOException at the moment.

A good solution that would allow registering with the correct email address again, without intervention of the web committee would be as follows:
Check if entry in NewUser exists for given membership ID, if so remove and recreate this entry correctly.
Resend registration email.

### PR DESCRIPTION
**Pull request by Koen1999**
This fixes #922

### CODE REVIEW

File: module/User/src/User/Model/NewUser.php [hunk 1]
```diff
@@ -72,6 +79,16 @@ public function getLidnr()
...
+    /**
+     * Set the user's membership number.
+     *
+     * @param int $lidnr
+     */
+    public function setLidnr($lidnr)
+    {
+        $this->lidnr = $lidnr;
+    }
+
```
Comment (kokx) [inline module/User/src/User/Model/NewUser.php]: Why all this useless rearrangement of methods? It completely obscures whatever you're actually changing.

**Comment by Koen1999** [comment]
I will attempt to undo the automatic code cleanup later this week so you'll be able to review it.

**Comment by Koen1999** [comment]
> 
> 
> I will attempt to undo the automatic code cleanup later this week so you'll be able to review it.

This became slightly later than expected, but I undid the reformatting of the code.

File: module/User/src/User/Service/User.php [hunk 2]
```diff
@@ -98,9 +98,19 @@ public function register($data)
             return null;
         }
 
+        $newUser = $this->getNewUserMapper()->getByLidnr($data['lidnr']);
+        if (is_null($newUser)) {
+            $time = $newUser->getTime();
+            $requiredInterval = (new DateTime())->sub(new DateInterval('P1D'));
+            if ($time > $requiredInterval) {
+                $form->setError(RegisterForm::)
```
Comment (kokx) [inline module/User/src/User/Service/User.php]: This seems a bit weird? And with syntax errors even?

Comment (Koen1999) [reply module/User/src/User/Service/User.php]: Apparently I forgot to fill it in after creating the error.

File: module/User/src/User/Form/Register.php [hunk 3]
```diff
@@ -69,6 +69,13 @@ public function setError($error)
                 ]
             ]);
             break;
+        case self::ERROR_ALREADY_REGISTERED:
```
Comment (Koen1999) [inline module/User/src/User/Form/Register.php]: I know indentation is incorrect, but at least it's consistent with the other cases.

**Pull request merged**